### PR TITLE
Manage End Condition + Score Scene

### PR DESCRIPTION
**Pull request by Arthi-chaud**
Add End condition management + event
Add Score Scene + relations to other scenes

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Non-breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by GitBluub — APPROVED** [review]

File: sources/Runner/GameScene.cpp [hunk 1]
```diff
@@ -34,6 +36,10 @@ namespace BBM
 		scene->addEntity("camera")
 			.addComponent<PositionComponent>(8, 20, 7)
 			.addComponent<CameraComponent>(Vector3f(8, 0, 8));
+		scene->addEntity("Timer")
+			.addComponent<TimerComponent>(std::chrono::seconds(60), [](WAL::Entity &, WAL::Wal &) {
```
Comment (AnonymusRaccoon) [inline sources/Runner/GameScene.cpp:40]: 60 seconds is a bit low, you should set it to 5 minutes I think.

File: sources/Runner/ScoreScene.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#include "Model/Model.hpp"
+
+namespace RAY2D = RAY::Drawables::Drawables2D;
+namespace RAY3D = RAY::Drawables::Drawables3D;
+
+namespace BBM
+{
+	std::shared_ptr<WAL::Scene> Runner::loadScoreScene(WAL::Scene &gameScene)
+	{
+        auto scene = std::make_shared<WAL::Scene>();
```
Comment (AnonymusRaccoon) [inline sources/Runner/ScoreScene.cpp:23]: Indentation.

File: sources/System/EndCondition/EndConditionSystem.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	{}
+
+	void EndConditionSystem::onSelfUpdate()
+	{
+		unsigned int alivePlayersCount = 0;
+		auto &view = this->_wal.getScene()->view<ScoreComponent, HealthComponent>();
+
+		if (!view.size())
+			return;
+        for (auto & [_ , scoreComponent, healthComponent]: view)
```
Comment (AnonymusRaccoon) [inline sources/System/EndCondition/EndConditionSystem.cpp:20]: Indent

**Review by EternalRat — APPROVED** [review]

File: sources/Runner/GameScene.cpp (hunk 1 above)
Comment (Octopus773) [reply sources/Runner/GameScene.cpp:40]: 3 mins

**Review by Octopus773 — APPROVED** [review]

**Review by TrueBabyChaise — APPROVED** [review]

**Pull request merged**